core/server/services/members/middleware.js

```diff
@@ -182,6 +182,17 @@ const createSessionFromMagicLink = async function (req, res, next) {
             }
         }
 
+        if (action === 'signin') {
+            const referrer = req.query.referrer;
```

Comment:
Hannah has pointed out that this should be `?r=` because we already have that pattern in place for private site logins 😃